packages/navigator/src/browser/navigator-model.ts

```diff
@@ -186,4 +186,12 @@ export class FileNavigatorModel extends FileTreeModel {
                 node1.id.length >= node2.id.length ? node1 : node2
             ) : undefined;
     }
+
+    async collapseAll(raw: CompositeTreeNode): Promise<boolean> {
+        const node = raw || this.selectedNodes[0];
```

Comment:
Should not we rather remove selection handling from super class and change existing clients to update selection always. Another fix would be to select a node only if a currently selected node belonged to collapsed nodes. It won't leave a tree in the weird state then selected nodes are collapsed and rely on clients to resolve such issues.
But overriding `collapseAll` with another implementation for each tree does not sound good. I see the same was already done for the output tree model.

Reply:
I looked at `tree-model.ts` and thought it was weird that the `collapseAll` was performing a selection but I wasn’t sure what the reason might be and I went this approach in order to reduce the possibility of a breaking behaviour. I like the idea that the `collapseAll` simply collapses and it’s the clients responsibility to perform a selection update. I think we will likely need to revisit other areas which do the same in an attempt to modify the base behaviour.

Reply:
we could limit this PR to fixing `collapseAll`

Reply:
Do you mean `collapseAll` for all trees via the `tree-model.ts` or limit the PR to fixing `collapseAll` for the Explorer?

Reply:
for all trees